### PR TITLE
Rename ClassOrganization #*category* into #*protocolName*

### PR DESCRIPTION
**Pull request by jecisc**
In the past #category was used to name Protocols. Now we use the name "Protocol".

This PR rename some method using "category" in their name to use "Protocol" instead

### CODE REVIEW

**Comment by jecisc** [comment]
GitHub Rate limiting for CI (yet again) #12285 

I hit that 2 times here

**Comment by MarcusDenker** [comment]
There are now conflicts

**Comment by MarcusDenker** [comment]
- [CI] test runs sometimes aborted due to SubscriptOutOfBounds error #12502

File: src/Kernel/ClassDescription.class.st [hunk 1]
```diff
@@ -137,9 +126,9 @@ ClassDescription >> allProtocolsUpTo: mostGenericClass [
 	lowercaseSortBlock  := [ :a :b | a asLowercase <= b asLowercase ].
 
 	(self allSuperclassesIncluding: mostGenericClass) do: [ :aClass |
-		 otherClassCategories addAll: aClass organization categories ].
+		 otherClassCategories addAll: aClass organization protocolsNames ].
 	otherClassCategories remove: 'no messages' ifAbsent: [  ].
-	thisClassCategories := self organization categories sorted: lowercaseSortBlock.
+	thisClassCategories := self organization protocolsNames sorted: lowercaseSortBlock.
```
Comment (demarey) [inline src/Kernel/ClassDescription.class.st]: maybe use `protocolNames`instead of `protocolsNames`

File: src/Kernel/ClassDescription.class.st [hunk 2]
```diff
@@ -503,7 +491,7 @@ ClassDescription >> copyAllCategoriesFrom: aClass [
 	these protocols into the method dictionary of the receiver, classified
 	under the appropriate protocol."
 
-	aClass organization categories do: [ :protocolName | self copyCategory: protocolName from: aClass ]
+	aClass organization protocolsNames do: [ :protocolName | self copyCategory: protocolName from: aClass ]
```
Comment (demarey) [inline src/Kernel/ClassDescription.class.st]: maybe use `protocolNames`instead of `protocolsNames`

**Review by demarey — APPROVED** [review]

**Comment by jecisc** [comment]
I took the review into account

**Pull request merged**